backend/src/nodes/impl/external_stable_diffusion.py

```diff
@@ -0,0 +1,96 @@
+import requests
+import base64
+import cv2
+import io
+import numpy as np
+import os
+from PIL import Image
+from enum import Enum
+from typing import Dict, Union
+from sanic.log import logger
+
+from .image_utils import normalize
+from ..utils.utils import get_h_w_c
+
+STABLE_DIFFUSION_HOST = os.environ.get("STABLE_DIFFUSION_HOST", "127.0.0.1")
+STABLE_DIFFUSION_PORT = os.environ.get("STABLE_DIFFUSION_PORT", "7860")
+
+STABLE_DIFFUSION_TEXT2IMG_URL = (
+    f"http://{STABLE_DIFFUSION_HOST}:{STABLE_DIFFUSION_PORT}/sdapi/v1/txt2img"
+)
+STABLE_DIFFUSION_IMG2IMG_URL = (
+    f"http://{STABLE_DIFFUSION_HOST}:{STABLE_DIFFUSION_PORT}/sdapi/v1/img2img"
+)
+STABLE_DIFFUSION_INTERROGATE_URL = (
+    f"http://{STABLE_DIFFUSION_HOST}:{STABLE_DIFFUSION_PORT}/sdapi/v1/interrogate"
+)
+STABLE_DIFFUSION_OPTIONS_URL = (
+    f"http://{STABLE_DIFFUSION_HOST}:{STABLE_DIFFUSION_PORT}/sdapi/v1/options"
+)
+
+
+def get(url) -> Dict:
+    response = requests.get(url)
+    return response.json()
+
+
+def post(url, json_data: Dict) -> Dict:
+    response = requests.post(url, json=json_data)
+    return response.json()
+
+
+# Call the API at import time
+# If this fails (because the API isn't up) it will raise an exception and the
+# nodes importing this file won't be registered
+STABLE_DIFFUSION_OPTIONS = get(STABLE_DIFFUSION_OPTIONS_URL)
```

Comment:
I wonder if this is a good idea. When I initially checked out this PR, the SD nodes didn't even show up because it couldn't connect to the options URL for whatever reason. I'm also not sure whether connecting to this URL once is a reliable indicator for the availability of the service. The nodes have to handle network errors anyway, since the device might disconnect for the network (laptop).

Hiding nodes based on factors outside the user's machine also presents usability issues. Imagine a user creating a chain with SD yesterday. Today the SD service has maintenance and so it refuses connections. Suddenly, the chain the user create but yesterday fails to load as chainner doesn't recognize the SD nodes anymore.

Yeah, I think having an error explaining that the service is unavailable at the moment would be preferable.

Reply:
It's not outside the user's machine though, this is a local server it's connecting to.

And my biggest thing was just not confusing people. We hide pytorch nodes if pytorch isn't installed, i thought this should be similar.

Reply:
Oh, you're right. I didn't even look at the host address and just assumed it was some remote server.